omod/src/main/java/org/openmrs/web/controller/LoginController.java

```diff
@@ -43,9 +43,10 @@ public String handleRequest(WebRequest webRequest, ModelMap model) {
 			}
 			webRequest.removeAttribute(WebConstants.INSUFFICIENT_PRIVILEGES, WebRequest.SCOPE_SESSION);
 		}
-		
-		//If there is a currently logged in user and they failed a privilege check, else go to login in page
-		if (Context.getAuthenticatedUser() != null && failedPrivilegeCheck) {
+		//If the user is logged in and there are no privilege problems, he should be redirected to home
+		if(Context.getAuthenticatedUser() != null && !failedPrivilegeCheck) {
+			return "/module/legacyui"+"/index";
```

Comment:
Sorry, my mistake, it should be "redirect:" not forward. Basically you don't want the URL to keep saying "login.htm" but you happen to display the home screen. Rather if they try to go to the login screen, but they are already logged in, then you should _redirect_ them so that the browser loads the home page instead.